### PR TITLE
DNN-18172: clean build warnings.

### PR DESCRIPTION
**Pull request by zyhfish**

### CODE REVIEW

File: DNN Platform/Admin Modules/Dnn.Modules.Console/web.config [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0"?>
+<configuration>
+	<runtime>
+		<assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
+			<dependentAssembly>
+				<assemblyIdentity name="Newtonsoft.Json" publicKeyToken="30AD4FE6B2A6AEED" culture="neutral"/>
+				<bindingRedirect oldVersion="0.0.0.0-10.0.0.0" newVersion="10.0.0.0"/>
```
Comment (tpluscode) [inline DNN Platform/Admin Modules/Dnn.Modules.Console/web.config]: No harm in having binding redirects but I looked at the projects and it turns out that we only have version 10 installed of Newtonsoft.Json. Why are there warnings in the first place?

File: DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj [hunk 2]
```diff
@@ -29,7 +29,7 @@
     <WarningLevel>4</WarningLevel>
     <CodeAnalysisRuleSet>AllRules.ruleset</CodeAnalysisRuleSet>
     <DocumentationFile>bin\ClientDependency.Core.XML</DocumentationFile>
-    <NoWarn>1591,1573</NoWarn>
+    <NoWarn>1591,1573,618</NoWarn>
```
Comment (tpluscode) [inline DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj]: I don't think we should be ignoring obsolete warnings. Any obsolete code should have a recommended replacement, right? If there is non I think we should keep the warnings

File: DNN Platform/Dnn.AuthServices.Jwt/Dnn.Jwt.dnn [hunk 3]
```diff
@@ -31,12 +31,14 @@
         </component>
         <component type="Script">
           <scripts>
-            <basePath>DesktopModules\AuthenticationServices\JWTAuth\Data\Scripts</basePath>
+            <basePath>DesktopModules\AuthenticationServices\JWTAuth</basePath>
```
Comment (tpluscode) [inline DNN Platform/Dnn.AuthServices.Jwt/Dnn.Jwt.dnn]: Oh, how is this related to build warnings? 🙂

File: DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj (hunk 2 above)
Comment (zyhfish) [reply DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj]: These warnings are comes from ClientDependency.Core project, and it contains several obsoleted warnings, if not ignore this maybe we can't reach the goal "less than 10 warnings.".

File: DNN Platform/Dnn.AuthServices.Jwt/Dnn.Jwt.dnn (hunk 3 above)
Comment (zyhfish) [reply DNN Platform/Dnn.AuthServices.Jwt/Dnn.Jwt.dnn]: The build scripts will try to copy these files from project directory(without basePath defined), and throw out warnings because these files aren't there.

File: DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj (hunk 2 above)
Comment (tpluscode) [reply DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj]: So these are actually coming from the 3rd party code we copied to DNN, right?

File: DNN Platform/Modules/MemberDirectory/DotNetNuke.Modules.MemberDirectory.csproj [hunk 4]
```diff
@@ -37,7 +37,7 @@
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
     <DocumentationFile>bin\DotNetNuke.Modules.MemberDirectory.XML</DocumentationFile>
-    <NoWarn>1591</NoWarn>
+    <NoWarn>1591,618</NoWarn>
```
Comment (tpluscode) [inline DNN Platform/Modules/MemberDirectory/DotNetNuke.Modules.MemberDirectory.csproj]: And what about the obsolete warnings in this project?

File: DNN Platform/Admin Modules/Dnn.Modules.Console/web.config (hunk 1 above)
Comment (zyhfish) [reply DNN Platform/Admin Modules/Dnn.Modules.Console/web.config]: this warning should be caused by there have old Newtonsoft.Json assembly installed in system level.

File: DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj (hunk 2 above)
Comment (zyhfish) [reply DNN Platform/Components/ClientDependency/Source/ClientDependency.Core.csproj]: Yep.
